src/lib.rs

```diff
@@ -586,6 +587,26 @@ fn readline_edit<H: Helper>(
                 s.edit_move_end()?;
                 if s.validate()? {
                     break;
+                } else {
+                    s.edit_insert('\n', 1)?;
+                }
+                continue;
+            }
+            Cmd::AcceptOrInsertLine => {
+                #[cfg(test)]
+                {
+                    editor.term.cursor = s.layout.cursor.col;
+                }
+                if s.has_hint() || !s.is_default_prompt() {
+                    // Force a refresh without hints to leave the previous
+                    // line as the user typed it after a newline.
+                    s.refresh_line_with_msg(None)?;
+                }
+                // Only accept value if cursor is at the end of the buffer
+                if s.validate()? && s.line.is_end_of_input() {
```

Comment:
I am not sure but we don't need to duplicate code:
```rust
Cmd::AcceptLine | Cmd::AcceptOrInsertLine => {
                #[cfg(test)]
                {
                    editor.term.cursor = s.layout.cursor.col;
                }
                // Accept the line regardless of where the cursor is.
                if s.has_hint() || !s.is_default_prompt() {
                    // Force a refresh without hints to leave the previous
                    // line as the user typed it after a newline.
                    s.refresh_line_with_msg(None)?;
                }
                 s.edit_move_end()?;
                 if s.validate()? && (cmd == Cmd::AcceptLine || s.line.is_end_of_input()) {
                     break;
                 } else {
                     s.edit_insert('\n', 1)?;
                 }
                 continue;
}
```

Reply:
Okay